contributors/design-proposals/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,162 @@
+| PriorityClassNameNotIn:priorityName1:priorityName2 | Match `kind=Pod` where `spec.priorityClassName notin (priorityName1, priorityName2)` |
+| PriorityClassNameExists |  Match `kind=Pod` where `spec.priorityClassName` |
+| PriorityClassNameNotExists |  Match `kind=Pod` where `!spec.priorityClassName` |
+
+*Example: Match all pods independent of priority*
+
+`$ kubectl create quota example --hard=count/pods=10`
+
+*Example: Match all pods where priority is foo*
+
+`$ kubectl create quota example --hard=count/pods=10 --scopes=PriorityClassNameIn:foo`
+
+*Example: Match all pods where there is no priority*
+
+`$ kubectl create quota example --hard=count/pods=10 --scopes=PriorityClassNameNotExists`
+
+To restrict usage of a particular priority without explicit quota, we can do the following:
+
+*Example: Any pod with priority cluster-services, or other-important-stuff must require explicit quota*
+
+```yaml
+apiVersion: apiserver.k8s.io/v1alpha1
+kind: AdmissionConfiguration
+plugins:
+- name: "ResourceQuota"
+  configuration:
+  apiVersion: resourcequota.admission.k8s.io/v1alpha1
+  kind: Configuration
+  limitedResources:
+  - resource: pods
+    matchScopes:
+    - "PriorityClassNameIn:cluster-services:other-important-stuff"
+```
+
+*Example: Any pod with priority set must require explicit quota*
+
+```yaml
+apiVersion: apiserver.k8s.io/v1alpha1
+kind: AdmissionConfiguration
+plugins:
+- name: "ResourceQuota"
+  configuration:
+  apiVersion: resourcequota.admission.k8s.io/v1alpha1
+  kind: Configuration
+  limitedResources:
+  - resource: pods
+    matchScopes:
+    - "PriorityClassNameExists"
+```
+
+There are some details to note:
+
+1. The absence of a quota for a particular priority class name means the quota is unbounded (unless configured otherwise by operator in admission control configuration).
+2. If multiple `ResourceQuota` apply to a Pod, the pod must satisfy all of them.
+
+
+### Expected behavior of ResourceQuota admission controller
+
+* We do not enforce referential integrity across objects. i.e. Creation or update of `ResourceQuota` object which `Scopes` names a `PriorityClass` that does not exist are allowed
+* `ResourceQuota` admission controller **should not reject** deletion of `ResourceQuota` objects with reference.
+
+#### Changes in AdmissionConfiguration
+
+It is configurable for `ResourceQuota` admission controller to reject pod with priority set but has no matched `ResourceQuota` found or not. This is mainly for backward compatibility consideration.
+
+This requires an extra boolean option in `AdmissionConfiguration`:
+```go
+// LimitedResource matches a resource whose consumption is limited by default.
+// To consume the resource, there must exist an associated quota that limits
+// its consumption.
+type LimitedResource struct {
+  APIGroup string `json:"apiGroup,omitempty"`
+
+  Resource string `json:"resource"`
+
+  MatchContains []string
+
+  // DefaultUnbound defines whether admission controller will or
```

Comment:
Since `matchScopes` are being newly added in this proposal, will be better to include `matchScope` in all the eamples of `LimitedResource`.